mmv1/third_party/terraform/resources/resource_apigee_env_keystore_alias_pkcs12.go

```diff
@@ -0,0 +1,627 @@
+package google
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"log"
+	"mime/multipart"
+	"os"
+	"reflect"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func ResourceApigeeEnvKeystoreAliasPkcs12() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceApigeeEnvKeystoreAliasPkcs12Create,
+		Read:   resourceApigeeEnvKeystoreAliasPkcs12Read,
+		Delete: resourceApigeeEnvKeystoreAliasPkcs12Delete,
+
+		Importer: &schema.ResourceImporter{
+			State: resourceApigeeEnvKeystoreAliasPkcs12Import,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(20 * time.Minute),
+			Delete: schema.DefaultTimeout(20 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"alias": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Alias Name`,
+			},
+			"file": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Cert content`,
+			},
+			"environment": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Required:    true,
+				Description: `Environment associated with the alias`,
+			},
+			"keystore": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Keystore Name`,
+			},
+			"org_id": {
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Required:    true,
+				Description: `Organization ID associated with the alias`,
+			},
+			"filehash": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Hash of the pkcs file",
+			},
+			"certs_info": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Computed:    true,
```

Comment:
```suggestion
				Computed:    true,
```
Discussed offline: sounds like `certs_info` and all its subfields are output-only, please change them from optional+computed to computed. Also please change other fields if they're also output-only.
Optional+computed in Terraform: still a field that user can specify, but it will be treated as output-only field if users does not specify explicitly.